src/python/pants/build_graph/build_configuration.py

```diff
@@ -17,23 +19,23 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass
 class BuildConfiguration:
     """Stores the types and helper functions exposed to BUILD files."""
 
+    _target_by_alias: Dict[Any, Any] = field(default_factory=dict)
```

Comment:
What's the difference between these?
```python
 _target_by_alias: Dict[Any, Any] = field(default_factory=dict)
```

```python
 _target_by_alias: Dict[Any, Any] = {}
```

If there's no difference, the latter is simpler and more idiomatic.

Reply:
The `field` thing is a weird thing that you have to do with `dataclass`es to set a dictionary as the default value. I think this has something to do with mutability, but I'm not entirely sure why it's necessary either.

Reply:
Huh, interesting: `ValueError: mutable default <class 'dict'> for field foo is not allowed: use default_factory`. TIL.